browser/app/js/buckets/Bucket.js

```diff
@@ -17,29 +17,101 @@
 import React from "react"
 import classNames from "classnames"
 import BucketDropdown from "./BucketDropdown"
+import ClickOutHandler from "react-onclickout"
+import * as actionsCommon from "../browser/actions"
+import { connect } from "react-redux"
 
-export const Bucket = ({ bucket, isActive, selectBucket }) => {
-  return (
-    <li
-      className={classNames({
-        active: isActive
-      })}
-      onClick={e => {
-        e.preventDefault()
-        selectBucket(bucket)
-      }}
-    >
-      <a
-        href=""
+export class Bucket extends React.Component {
+  constructor(props) {
+    super(props)
+    this.state = {
+      bucketDropdownActive: false,
+      bucketDropdownEnters: false
+    }
+  }
+
+  showBucketDropdown() {
+    this.setState({
+      bucketDropdownActive: true
+    })
+
+    setTimeout(() => {
+      this.setState({
+        bucketDropdownEnters: true
+      })
+    }, 1)
+  }
+
+  hideBucketDropdown() {
+    this.setState({
+      bucketDropdownEnters: false
+    })
+
+    setTimeout(() => {
+      this.setState({
+        bucketDropdownActive: false
+      })
+    }, 300)
+  }
+
+  hideBucketDropdownOnEscape(e) {
+    if (e.keyCode === 27) {
+      this.hideBucketDropdown()
+    }
+  }
+
+  render() {
+    const { bucket, isActive, selectBucket, closeSidebar } = this.props
+
+    return (
+      <div
         className={classNames({
-          "fesli-loading": false
+          buckets__item: true,
+          "buckets__item--active": isActive,
+          "buckets__item--enters": this.state.bucketDropdownEnters
         })}
+        onClick={e => {
+          e.preventDefault()
+          this.hideBucketDropdown.bind(this)
+          setTimeout(() => {
+            selectBucket(bucket)
+          })
+
+          if (e.target.className != "buckets__toggle") {
```

Comment:
I see this condition is added to check whether the user has clicked on the bucket name or on the dropdown caret. You can actually stop the event from bubbling to the parent, so we can avoid this condition.

the code would become something like following
```
<i
  tabIndex="0"
  className="buckets__toggle"
  onClick={e => {
    e.stopPropagation()
    this.state.bucketDropdownActive
      ? this.hideBucketDropdown()
      : this.showBucketDropdown()
  }}
  onKeyDown={this.hideBucketDropdownOnEscape.bind(this)}
/>
```
once we call `e.stopPropagation()`, the outer element will not receive the click event